zk/stages/stage_sequence_execute.go

```diff
@@ -742,6 +742,17 @@ func sequencingBatchStep(
 		if err != nil || needsUnwind {
 			return err
 		}
+
+		if _, err := rawdb.IncrementStateVersionByBlockNumberIfNeeded(batchContext.sdb.tx, block.NumberU64()); err != nil {
```

Comment:
Will we get a crash on block==nil again here since you have removed the check?

Reply:
We're now inside the block loop rather than outside of it when writing the state version, we've switched to doing this per block rather than per batch. the only reason it could be nil is if we missed an error check above, we also use block in other calls above in much the same way.